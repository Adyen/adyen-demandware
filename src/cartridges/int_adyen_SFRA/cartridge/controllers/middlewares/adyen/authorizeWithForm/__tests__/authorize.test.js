jest.mock('../order');
jest.mock('../error');
jest.mock('../payment');

const OrderMgr = require('dw/order/OrderMgr');
const COHelpers = require('*/cartridge/scripts/checkout/checkoutHelpers');
const handleAuthorize = require('../authorize');
const handleOrderConfirmation = require('../order');
const handleInvalidPayment = require('../payment');
const handleError = require('../error');

let req;
let res;
beforeEach(() => {
  jest.clearAllMocks();
  req = { form: { MD: 'mocked_md' } };
  res = { redirect: jest.fn() };
});

afterEach(() => {
  jest.resetModules();
});

describe('Authorize', () => {
  it('should authorize when MD is valid', () => {
    OrderMgr.getPaymentInstruments.mockImplementation(() => [
      { custom: { adyenPaymentData: 'Authorised' } },
    ]);
    window.session.privacy.MD = 'mocked_md';
    handleAuthorize({ req, res, next: jest.fn() });
    expect(handleOrderConfirmation.mock.calls).toMatchSnapshot();
  });
  it('should handle error when MD is invalid', () => {
    window.session.privacy.MD = 'invalid_mocked_md';
    handleAuthorize({ req, res, next: jest.fn() });
    expect(handleError.mock.calls).toMatchSnapshot();
  });
  it('should handle invalid payment when result code is not Authorised', () => {
    OrderMgr.getPaymentInstruments.mockImplementation(() => [
      { custom: { adyenPaymentData: 'Not_Authorised' } },
    ]);
    window.session.privacy.MD = 'mocked_md';
    handleAuthorize({ req, res, next: jest.fn() });
    expect(handleInvalidPayment.mock.calls).toMatchSnapshot();
  });
  it('should handle invalid payment when there is an error while placing an order', () => {
    OrderMgr.getPaymentInstruments.mockImplementation(() => [
      { custom: { adyenPaymentData: 'Authorised' } },
    ]);
    COHelpers.placeOrder.mockImplementation(() => ({ error: true }));
    window.session.privacy.MD = 'mocked_md';
    handleAuthorize({ req, res, next: jest.fn() });
    expect(handleInvalidPayment.mock.calls).toMatchSnapshot();
  });
});
