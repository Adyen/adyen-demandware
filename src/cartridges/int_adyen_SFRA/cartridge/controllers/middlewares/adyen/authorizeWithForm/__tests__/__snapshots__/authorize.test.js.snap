// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Authorize should authorize when MD is valid 1`] = `
Array [
  Array [
    Object {
      "custom": Object {
        "adyenPaymentData": null,
      },
    },
    Object {
      "resultCode": "Authorised",
    },
    Object {
      "custom": Object {
        "Adyen_pspReference": "mocked_pspRef",
      },
      "getPaymentInstruments": [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Array [
              Object {
                "custom": Object {
                  "adyenPaymentData": null,
                },
              },
            ],
          },
        ],
      },
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "setExportStatus": [MockFunction],
      "setPaymentStatus": [MockFunction],
    },
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "MD": "mocked_md",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;

exports[`Authorize should handle error when MD is invalid 1`] = `
Array [
  Array [
    "Session variable does not exists",
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "MD": "mocked_md",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;

exports[`Authorize should handle invalid payment when result code is not Authorised 1`] = `
Array [
  Array [
    Object {
      "custom": Object {
        "Adyen_pspReference": "mocked_pspRef",
      },
      "getPaymentInstruments": [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Array [
              Object {
                "custom": Object {
                  "adyenPaymentData": null,
                },
              },
            ],
          },
        ],
      },
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "setExportStatus": [MockFunction],
      "setPaymentStatus": [MockFunction],
    },
    "payment",
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "MD": "mocked_md",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;

exports[`Authorize should handle invalid payment when there is an error while placing an order 1`] = `
Array [
  Array [
    Object {
      "custom": Object {
        "Adyen_pspReference": "mocked_pspRef",
      },
      "getPaymentInstruments": [MockFunction] {
        "calls": Array [
          Array [
            undefined,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": Array [
              Object {
                "custom": Object {
                  "adyenPaymentData": null,
                },
              },
            ],
          },
        ],
      },
      "orderNo": "mocked_orderNo",
      "orderToken": "mocked_orderToken",
      "setExportStatus": [MockFunction],
      "setPaymentStatus": [MockFunction],
    },
    "placeOrder",
    Object {
      "next": [MockFunction],
      "req": Object {
        "form": Object {
          "MD": "mocked_md",
        },
      },
      "res": Object {
        "redirect": [MockFunction],
      },
    },
  ],
]
`;
